src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp

```diff
@@ -18,31 +18,13 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
 
     IShapeInfer::Result infer(const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
                               [[maybe_unused]] const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
-        OPENVINO_ASSERT(input_shapes.size() >= 3,
-                        "SDPAShapeInfer: expected at least 3 inputs, got ",
-                        input_shapes.size());
+        auto inputs_size = input_shapes.size();
+        OPENVINO_ASSERT(inputs_size >= 3, "SDPAShapeInfer: expected at least 3 inputs, got ", inputs_size);
         const auto& query_dims = input_shapes.front().get();
         VectorDims present_v_dims = input_shapes.back().get();
-        const auto& beam_idx_dims = input_shapes[input_shapes.size() - 3].get();
-        const auto& permute_axes = m_config.permute_axes;
-
-        if (permute_axes.empty()) {
-            // [B, H, L, S]
-            present_v_dims[0] = beam_idx_dims[0];
-            present_v_dims[2] += query_dims[2];
-            // normal and fast path
-            if (present_v_dims[3] == query_dims[3]) {
-                return {{query_dims, present_v_dims, present_v_dims}, ShapeInferStatus::success};
-            }
-
-            // diff kv feature size
-            auto output_dims = query_dims;
-            output_dims[3] = present_v_dims[3];
-            auto present_k_dims = present_v_dims;
-            present_k_dims[3] = query_dims[3];
-            return {{output_dims, present_k_dims, present_v_dims}, ShapeInferStatus::success};
-        }
-
+        const auto& beam_idx_dims = input_shapes[inputs_size - 3].get();
+        const auto& permute_axes_origin = m_config.permute_axes;
+        auto permute_axes = (permute_axes_origin.empty()) ? std::vector<size_t>{0, 1, 2, 3} : permute_axes_origin;
```

Comment:
```suggestion
        static const std::vector<size_t> direct_axis_order = {0, 1, 2, 3};
        const auto& permute_axes = (permute_axes_origin.empty()) ? direct_axis_order : permute_axes_origin;
```
This allows to avoid a copy during instancing of the `permute_axes` object, which is in fact read only.
Also this allows to avoid a malloc, each time `permute_axes_origin.empty()` is `true`.

Reply:
fixed